dali/pipeline/operators/reader/loader/sequence_loader.cc

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <glob.h>
+#include <algorithm>
+#include <map>
+
+#include "dali/pipeline/operators/reader/loader/sequence_loader.h"
+#include "dali/util/file.h"
+
+namespace dali {
+
+std::string parent_dir(std::string path) {
+  size_t idx = path.rfind("/");
+  DALI_ENFORCE(idx != std::string::npos, "NO PARENT DIRECTORY FOR GIVEN PATH");
+  return path.substr(0, idx + 1);  // return with trailing "/"
+}
+
+void SequenceLoader::PrepareEmpty(TensorSequence *sequence) {
+  sequence->tensors.resize(sequence_length_);
+  for (auto &t : sequence->tensors) {
+    PrepareEmptyTensor(&t);
+  }
+}
+
+void SequenceLoader::ReadSample(TensorSequence *sequence) {
+  // TODO(klecki) this is written as a prototype for video handling
+  const auto &stream = streams_[current_stream_];
+  // TODO(klecki) we probably should buffer the "stream", or recently used
+  // frames
+  for (int i = 0; i < sequence_length_; i++) {
+    LoadFrame(stream, current_frame_ + i, &sequence->tensors[i]);
+  }
+  current_frame_++;
+  // wrap-around
+  if (current_frame_ == stream_sizes_[current_stream_]) {
+    current_stream_++;
+    current_frame_ = 0;
+  }
+  if (current_stream_ == streams_.size()) {
+    current_stream_ = 0;
+  }
+}
+
+Index SequenceLoader::Size() {
+  return total_size_;
+}
+
+std::vector<SequenceLoader::Stream> SequenceLoader::ParseStreams(string file_root) {
+  glob_t glob_buff;
+  std::string glob_pattern = file_root + "/*/*";
+  const int glob_flags = 0;
+  glob(glob_pattern.c_str(), glob_flags, nullptr, &glob_buff);
```

Comment:
Is filesystem dependency temporary? It might not be the best idea in production code

Reply:
Currently we want to assume some layout of directories for extracted frames. As we add support for video it will be a list of video streams.